### DISCUSSION TITLE
Blob for video, instead of url

### DISCUSSION THREAD

**Discussion by stanislavlysenko0912** [filamentphp/filament#10270] [Help]
Why im getting blob url for video preview, instead of full url.

<img width="411" alt="image" src="https://github.com/filamentphp/filament/assets/86171768/84e69d40-9ed7-4564-9a44-4917e07b15f3">

```php
Forms\Components\FileUpload::make('video')
                            ->label('Video')
                            ->required()
                            ->disk('public')
                            ->directory('short-videos')
                            ->visibility('public'),
```

In rendered admin, i see.
```html
<video controls="true" src="blob:https://mydomain.com/ca24e9cd-b749-4129-b507-e24984297136"></video>
```
If i open this url, i see 
<img width="367" alt="image" src="https://github.com/filamentphp/filament/assets/86171768/6399d16b-ad87-4c25-a2b3-321ae8fc2353">

DB Row
```json
[
  {
    "id": 2,
    "description": "{\"en\":\"test en\",\"es\":\"test es\"}",
    "video": "short-videos/01HHFWCYZZWM157Y1E8M07QMAC.mp4",
    "is_visible": true,
    "show_on_homepage": true,
    "created_at": null,
    "updated_at": null
  }
]
```

I can open video by
https://mydomain.com/short-videos/01HHFWCYZZWM157Y1E8M07QMAC.mp4

**Comment by lucacastelnuovo** [comment]
Does the video in the rendered admin work? If so I believe it is working as intended.
Blobs are not meant to access directly (https://stackoverflow.com/a/16179887).

**Reply by stanislavlysenko0912** [reply]
I want just use full url for file instead of blob in preview.
How i can do this?

Like in demo project

**Reply by lucacastelnuovo** [reply]
It is not necessary to solve this issue I think but I really like: https://filamentphp.com/plugins/filament-spatie-media-library

**Reply by stanislavlysenko0912** [reply]
Very thanks, but i tried. Dont helped.
Image preview loading good, problems with video 🥲

**Reply by lucacastelnuovo** [reply]
Could you describe:
- The problem you are experiencing
- What you did to try and solve it

**Reply by aytackayin** [reply]
Preview works correctly in Filament 2.17.8, but it doesn't work in Filament 3.1.34. The issue seems to be that when generating the Blob URL, it doesn't recognize video and mp3 files, resulting in an empty Blob. It appears that the Blob URL is generated using JavaScript, and there may be an issue in the file-upload.js file. In the older version, FilePond 4.30.4 is used, while the newer version uses 4.30.12.

As a solution, you can write JavaScript code to replace the video's src with the file URL. It seems that they don't consider this problem as a bug. In my opinion, it could be a bug.

**Comment by phpust** [comment]
i found a solution. problem is inside file-upload.js.
there is a method called getFiles.
```
getFiles: async function () {
            await this.getUploadedFiles()

            let files = []

            for (const uploadedFile of Object.values(this.fileKeyIndex)) {
                if (!uploadedFile) {
                    continue
                }

                files.push({
                    source: uploadedFile.url,
                    options: {
                        type: 'local',
                        ...(!uploadedFile.type ||
                        /^image/.test(uploadedFile.type)
                            ? {}
                            : {
                                  file: {
                                      name: uploadedFile.name,
                                      size: uploadedFile.size,
                                      type: uploadedFile.type,
                                  },
                              }),
                    },
                })
            }

            return shouldAppendFiles ? files : files.reverse()
        },
```
it must change into
```getFiles: async function () {
            await this.getUploadedFiles()

            let files = []

            for (const uploadedFile of Object.values(this.fileKeyIndex)) {
                if (!uploadedFile) {
                    continue
                }

                files.push({
                    source: uploadedFile.url,
                    options: {
                        type: 'local',
                        ...(!uploadedFile.type ||
                        /^image/.test(uploadedFile.type)
                            ? {}
                            : {
                                  file: {
                                      name: uploadedFile.name,
                                      size: uploadedFile.size,
                                      type: uploadedFile.type,
                                      url: uploadedFile.url,
                                      mock: true, // Should be set to true to avoid full file preloading before showing preview
                                  },
                              }),
                    },
                })
            }

            return shouldAppendFiles ? files : files.reverse()
        },```
if you published your assets, then the file is located at public/js/filament/forms/components/file-upload.js

**Reply by phpust** [reply]
@danharrin please check if this is a valid approach

**Reply by aliteleb** [reply]
Have you tried
`->fetchFileInformation(false)`

**Reply by phpust** [reply]
> Have you tried `->fetchFileInformation(false)`

works perfectly. :)
